### DISCUSSION TITLE
Error with AWS Amplify in Ignite React Native Boilerplate Using Hermes: 'secure crypto unusable, insecure Math.random not allowed

### DISCUSSION THREAD

**Discussion by abdulmateentechbits** [infinitered/ignite#2496] [Q&A]
Hello everyone,

I am encountering an issue when trying to integrate AWS Amplify into the Ignite React Native boilerplate. Specifically, I am facing the following error:

```
**ERROR  Error: secure crypto unusable, insecure Math.random not allowed, js engine: hermes
LOG  Running "awsAmplifyAuthentication" with {"rootTag":11}
ERROR  Invariant Violation: "awsAmplifyAuthentication" has not been registered. This can happen if:
* Metro (the local dev server) is run from the wrong folder. Check if Metro is running, stop it and restart it in the current project.
* A module failed to load due to an error and `AppRegistry.registerComponent` wasn't called., js engine: hermes**
```

Interestingly, AWS Amplify authentication works perfectly fine in a simple React Native app. I have ensured that I am using the correct dependencies and configurations, but this issue persists when working with the Ignite boilerplate.

**Question:**

Has anyone else encountered a similar issue when using AWS Amplify with the Ignite React Native boilerplate, particularly with the Hermes JavaScript engine? If so, could you please share any insights or solutions to resolve this problem?

**Your assistance is greatly appreciated. Thank you!**

**Comment by jamonholmgren** [comment]
@abdulmateentechbits Thanks for your well-formed discussion!

Could you give steps to replicate this in a minimal Ignite project?

**Reply by Abdulmateenchitrali** [reply]
@jamonholmgren 
Thank you for your reply, and I appreciate your willingness to assist. To clarify, I followed these steps:

I began by following the instructions in this documentation:

[Getting Started with AWS Amplify in React Native](https://docs.amplify.aws/start/getting-started/setup/q/integration/react-native/)
After completing the initial setup, I proceeded to implement authentication using AWS Amplify by following this documentation:

[Authentication Setup with AWS Amplify in React Native](https://docs.amplify.aws/start/getting-started/auth/q/integration/react-native/)
It was at this point that I encountered an error when wrapping the Higher-Order Component (HOC) withAuthenticator. The error occurred in the following code section

```
/* eslint-disable import/first */
/**
 * Welcome to the main entry point of the app. In this file, we'll
 * be kicking off our app.
 *
 * Most of this file is boilerplate and you shouldn't need to modify
 * it very often. But take some time to look through and understand
 * what is going on here.
 *
 * The app navigation resides in ./app/navigators, so head over there
 * if you're interested in adding screens and navigators.
 */
if (__DEV__) {
  // Load Reactotron configuration in development. We don't want to
  // include this in our production bundle, so we are using `if (__DEV__)`
  // to only execute this in development.
  require("./devtools/ReactotronConfig.ts")
}
import "./i18n"
import "./utils/ignoreWarnings"
import { useFonts } from "expo-font"
import React from "react"
import { initialWindowMetrics, SafeAreaProvider } from "react-native-safe-area-context"
import * as Linking from "expo-linking"
import { useInitialRootStore } from "./models"
import { AppNavigator, useNavigationPersistence } from "./navigators"
import { ErrorBoundary } from "./screens/ErrorScreen/ErrorBoundary"
import * as storage from "./utils/storage"
import { customFontsToLoad } from "./theme"
import Config from "./config"
import { Amplify } from 'aws-amplify';
import awsExports from '../src/aws-exports';
import { withAuthenticator } from 'aws-amplify-react-native';

Amplify.configure(awsExports);

export const NAVIGATION_PERSISTENCE_KEY = "NAVIGATION_STATE"

// Web linking configuration
const prefix = Linking.createURL("/")
const config = {
  screens: {
    Login: {
      path: "",
    },
    Welcome: "welcome",
    Demo: {
      screens: {
        DemoShowroom: {
          path: "showroom/:queryIndex?/:itemIndex?",
        },
        DemoDebug: "debug",
        DemoPodcastList: "podcast",
        DemoCommunity: "community",
      },
    },
  },
}

interface AppProps {
  hideSplashScreen: () => Promise<void>
}

/**
 * ======================================================>App component goes here
 */

export default withAuthenticator(App) // ========================> this cause the error
```

**Comment by frankcalise** [comment]
Have you read through this previous issue? https://github.com/infinitered/ignite/issues/1939 

Might help you some
